src/sentry/models/group.py

```diff
@@ -377,6 +377,39 @@ def get_latest_event(self):
                 self._latest_event = None
         return self._latest_event
 
+    def get_latest_event_for_environments(self, environments=[]):
+        from sentry.utils import snuba
+        from sentry.models import SnubaEvent
+
+        use_snuba = options.get('snuba.events-queries.enabled')
+
+        # Fetch without environment if Snuba is not enabled
+        if not use_snuba:
+            return self.get_latest_event()
+
+        conditions = []
+
+        if len(environments) > 0:
+            conditions.append(['environment', 'IN', environments])
+
+        result = snuba.raw_query(
+            start=datetime.utcfromtimestamp(0),
+            end=datetime.utcnow(),
+            selected_columns=SnubaEvent.selected_columns,
+            conditions=conditions,
+            filter_keys={
+                'issue': [self.id],
+                'project_id': [self.project_id],
+            },
+            orderby=['-timestamp', '-event_id'],
+            limit=1,
+        )
+
+        if 'error' not in result and len(result['data']) == 1:
+            return SnubaEvent(result['data'][0])
```

Comment:
If we end up doing this pattern a lot:
```
result = snuba.raw_query(selected_columns=SnubaEvent.selected_columns, conditions=...)
event = SnubaEvent(result['data'][0])
```
then we should probably turn it into something we can call from a manager like `SnubaEvent.objects.filter(... conditions)`